ruby/advanced_ruby/pattern_matching.md

```diff
@@ -666,12 +666,14 @@ With pattern matching, we do need to consider the case if there is no match. Wit
 Pattern Matching provides a powerful way to get at the data you need in a way that makes sense. But as with any tool, you don't need to reach for it in every use case. For data structures that are shallow, just think about whether you need to bring in a pattern match when you could just grab the value using the hash and array methods already available. If you do find yourself deep in a nested structure with many checks for `nil` and a lot of edge case handling, it might be time to see if pattern matching brings something to the table.
 
 ### Assignment
+
 <div class="lesson-content__panel" markdown="1">
 
-1. Read the [Ruby docs](https://docs.ruby-lang.org/en/3.3/syntax/pattern_matching_rdoc.html) on pattern matching. They cover quite a lot of ground.
+1. Read the [Ruby docs on pattern matching](https://docs.ruby-lang.org/en/3.3/syntax/pattern_matching_rdoc.html).
```

Comment:
Only need to replace the first sentence, no need to chop off the rest of the line.
```suggestion
1. Read the [Ruby docs on pattern matching](https://docs.ruby-lang.org/en/3.3/syntax/pattern_matching_rdoc.html). They cover quite a lot of ground.
```

Reply:
Sure thing :)
Am i supposed to create a new PR after changing the files , or is there a way to edit this one ?

Reply:
The markdownlint VS Code extension is catching lint errors in the concerned .md files , but the npm run fix:lesson -- "./path/to/lesson" commands dont seem to work , they show the following output in the CLI : 
Finding: ./ruby/advanced_ruby/pattern_matching.md
Linting: 1 file(s)
Summary: 0 error(s)
and make no changes whatsoever in the .md files.

Reply:
100% certain that no changes were made to the `pattern_matching.md` file at all?

Given the markdown from your PR at the moment, running 
```bash
npm run fix:lesson -- "./ruby/advanced_ruby/pattern_matching.md"
```
will modify the `pattern_matching.md` file (with the line breaks and link type conversions at the end of the file) then report exactly what you've shown (because there won't be any errors left after fixing).

Can you confirm that the `pattern_matching.md` file is actually untouched? i.e. the knowledge check links are still using HTML anchor tags and not markdown links? And that the `</div>` tag just after the single assignment step is still directly after the list and not with a blank line inbetween?

Reply:
So yeah , they are not untouched , the </div> tag is getting a blank line before it and knowledge check links are now using markdown links , but these are still visible MD033/no-inline-html: Inline HTML [Element: div] , would they pose any problem ?

Reply:
Not sure why you're getting that since our markdownlint config file is has the stuff set up for the MD033 rule.

Make sure that your VSCode workspace is the curriculum repo. If the curriculum repo is nested within a different workspace folder, then the VSCode extension may not be picking up the correct config file, so reports something different to the npm package.

You can run the `lint:lesson` script to check if there are any errors. If not, push and it'll be fine here. If anything further needs doing then I can always let you know

Reply:
Sure

Reply:
I think i might have fixed the linting issues now . There was an error in project_custom_enumerable.md as well that expected ###Lesson overview but Actual was ###Assignment , i guess that has got something to do with the author not choosing to add the lesson overview , so i have not done anything about that , but do tell if anything else is required of me.

Reply:
The custom enumerables one is a project, which uses a different configuration that has a different lesson heading structure. You probably ran the `lint:lesson` script on the custom enumerable project, whereas if you ran `lint:project`, it'd be happy with 0 errors.

As you can see, the GitHub actions that run the automated lint checks are all happy with the files now - great job! :rocket:

Reply:
Oh i get it now !
Thanks sir for the detailed explanations and helping me in my first PR :)